lib/warnings.js

```diff
@@ -25,4 +25,6 @@ warning.create('FastifyDeprecation', 'FSTDEP006', 'You are decorating Request/Re
 
 warning.create('FastifyDeprecation', 'FSTDEP007', 'You are trying to set a HEAD route using "exposeHeadRoute" route flag when a sibling route is already set. See documentation for more info.')
 
+warning.create('FastifyDeprecation', 'FSTDEP008', 'You are using the legacy way to add route-level hooks. Use the one suggested in the documentation instead.')
```

Comment:
It is not clear to me what it means with "legacy way". could it be more straightforward to understand?